ksql-engine/src/test/resources/query-validation-tests/case-expression.json

```
@@ -0,0 +1,411 @@
+              "STATE": "STATE_5",
+              "STREET": "STREET_8",
+              "NUMBER": 380,
+              "ZIPCODE": 866
+            },
+            "ARRAYCOL": [
+              9.887072401304447,
+              5.217021497196517,
+              5.604857288119519,
+              4.628527278923561,
+              6.367135367927281
+            ]
+          },
+          "key": 101
+        },
+        {
+          "topic": "test_topic",
+          "timestamp": 0,
+          "value": {
+            "ORDERID": 5,
+            "ITEMID": {
+              "ITEMID": 5,
+              "CATEGORY": {
+                "ID": 1,
+                "NAME": "Produce"
+              },
+              "NAME": "Item_5"
+            },
+            "ORDERTIME": 1528224286568,
+            "MAPCOL": {
+              "key3": 4.9160189684727,
+              "key2": 1.3876747586974092,
+              "key1": 6.688854425726891
+            },
+            "ORDERUNITS": 1.0,
+            "ADDRESS": {
+              "CITY": "CITY_6",
+              "STATE": "STATE_3",
+              "STREET": "STREET_8",
+              "NUMBER": 294,
+              "ZIPCODE": 724
+            },
+            "ARRAYCOL": [
+              1.6856668854084866,
+              3.4970511301361484,
+              9.143163282671962,
+              2.196065628133206,
+              4.343961390870502
+            ]
+          },
+          "key": 101
+        }
+      ],
+      "outputs": [
+        {
+          "topic": "S1",
+          "timestamp": 0,
+          "value": {
+            "CASE_RESAULT": "medium"
+          },
+          "key": 0
+        },
+        {
+          "topic": "S1",
+          "timestamp": 0,
+          "value": {
+            "CASE_RESAULT": "large"
+          },
+          "key": 100
+        },
+        {
+          "topic": "S1",
+          "timestamp": 0,
+          "value": {
+            "CASE_RESAULT": "large"
+          },
+          "key": 101
+        },
+        {
+          "topic": "S1",
+          "timestamp": 0,
+          "value": {
+            "CASE_RESAULT": "medium"
+          },
+          "key": 101
+        },
+        {
+          "topic": "S1",
+          "timestamp": 0,
+          "value": {
+            "CASE_RESAULT": "small"
+          },
+          "key": 101
+        }
+      ]
+    },
+    {
+      "format": ["AVRO", "JSON"],
+      "statements": [
+        "CREATE STREAM orders (itemid STRUCT< ITEMID BIGINT, NAME VARCHAR, CATEGORY STRUCT< ID BIGINT, NAME VARCHAR>>, address STRUCT < number bigint, street varchar, city varchar, state varchar, zipcode bigint>) WITH (kafka_topic='test_topic', value_format='{FORMAT}');",
+        "CREATE STREAM S1 AS SELECT CASE WHEN ADDRESS -> STATE = 'STATE_1' THEN ITEMID -> CATEGORY -> NAME WHEN ADDRESS -> STATE = 'STATE_3' THEN ADDRESS -> CITY ELSE 'default' END AS case_resault FROM orders;"
+      ],
+      "name": "simple case expression with multiple expression and the same type",
```

```suggestion
      "name": "searched case expression with multiple expression",
```
Rename to avoid confusion if we want to support simple case expressions in the future. (Also simplify test name)

---

Changed it to `searched`. I intentionally made it complex so we also test the case expression with `STRUCT` type.